src/Lexer.php

```diff
@@ -21,7 +21,19 @@
  *
  * @author Konstantin Kudryashov <ever.zet@gmail.com>
  *
- * @phpstan-type TToken array{type: string, line: int, value: string|null, deferred: bool, tags?: list<string>, keyword?: string, keyword_type?: string, text?: string, indent?: int, columns?: list<string>}
+ * @phpstan-import-type TLanguage from KeywordsInterface
+ * @phpstan-import-type TKeywordsString from KeywordsInterface
+ * @phpstan-import-type TKeywordsType from KeywordsInterface
+ * @phpstan-import-type TStepKeywordsType from KeywordsInterface
+ * @phpstan-import-type TGeneralKeywordsType from KeywordsInterface
```

Comment:
Getting this `TGeneralKeywordsType` from KeywordsInterface is also creating a dependency that does not exist in practice. The token types are a concept of the Lexer, not of the KeywordsInterface.
A good proof is that you define this type alias in KeywordsInterface but you never actually use it in this file. This is a very good indication that the type alias is defined in a place where it does not belong.